src/pages/create/Opportunity.tsx

```diff
@@ -0,0 +1,182 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+import {
+  Button,
+  H0,
+  H2,
+  Stack,
+  useDeskproAppClient,
+  useInitialisedDeskproAppClient,
+} from "@deskpro/app-sdk";
+import { zodResolver } from "@hookform/resolvers/zod";
+import { useEffect, useState } from "react";
+import { useForm } from "react-hook-form";
+import { useNavigate, useParams } from "react-router-dom";
+import { z, ZodTypeAny } from "zod";
+
+import { getObjectMeta, postData } from "../../api/api";
+import { useQueryWithClient } from "../../hooks";
+import { QueryKey } from "../../query";
+import { getMetadataBasedSchema } from "../../schemas/default";
+import opportunityJson from "../../resources/default_layout/opportunity.json";
+import { FieldMappingInput } from "../../components/FieldMappingInput/FieldMappingInput";
+import { Field } from "../../api/types";
+
+const nonUsableFields = ["AccountId", "CreatedDate", "CreatedById"];
+
+const requiredFields = ["Name", "StageName", "CloseDate"];
+
+export const CreateOpportunity = () => {
+  const navigate = useNavigate();
+
+  const [submitting, setSubmitting] = useState(false);
+  const [schema, setSchema] = useState<ZodTypeAny>(z.object({}));
+  const { object, parentId } = useParams();
+
+  const { client } = useDeskproAppClient();
+  const {
+    register,
+    formState: { errors },
+    handleSubmit,
+    setValue,
+    watch,
+  } = useForm<any>({
+    resolver: zodResolver(schema),
+  });
+  useInitialisedDeskproAppClient((client) => {
+    client.deregisterElement("salesforcePlusButton");
+    client.setTitle("Add Opportunity");
+  });
+
+  useEffect(() => {
+    if (parentId) {
+      setValue(object as string, parentId);
+    }
+  }, [parentId, setValue, object]);
+
+  useEffect(() => {
+    register("StageName", {
```

Comment:
What is the purpose of these hard-coded field names? `StageName` and `CloseDate`